dipy/denoise/localpca.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+import scipy as sp
+import scipy.linalg as sla
+from warnings import warn
+
+# Try to get the SVD through direct API to lapack:
+try:
+    from scipy.linalg.lapack import sgesvd as svd
+    svd_args = [1, 0]
+# If you have an older version of scipy, we fall back
+# on the standard scipy SVD API:
+except ImportError:
+    from scipy.linalg import svd
+    svd_args = [False]
+
+
+def localpca(arr, sigma, patch_radius=1):
+    r"""Local PCA-based denoising of diffusion datasets.
+
+    Parameters
+    ----------
+    arr : 4D array
+        Array of data to be denoised.
```

Comment:
You are right! They don't need to be different, what I meant was that the last dimension should contain the diffusion-weighted signals.